File: Leetcode 3901-4000/3974/3974.py
# Leetcode 3974: Maximum Total Sum of K Selected Elements
# https://leetcode.com/problems/maximum-total-sum-of-k-selected-elements/
# Solved on 18th of July, 2026
class Solution:
    def maxSum(self, nums: list[int], k: int, mul: int) -> int:
        """
        Finds the maximum total sum of k selected elements from the array.

        Args:
            nums: The input array of integers.
            k: The number of elements to select.
            mul: The multiplier.

        Returns:
            The maximum total sum of k selected elements.
        """
        nums.sort(reverse=True)
        totalSum = 0

        for i in range(k):
            currentMul = max(1, mul - i)
            totalSum += nums[i] * currentMul
        
        return totalSum

File: Leetcode 3901-4000/3974/3974.java
// Leetcode 3974: Maximum Total Sum of K Selected Elements
// https://leetcode.com/problems/maximum-total-sum-of-k-selected-elements/
// Solved on 18th of July, 2026
import java.util.Arrays;

class Solution {
    /**
     * Finds the maximum total sum of k selected elements from the array.
     * 
     * @param nums The input array of integers.
     * @param k The number of elements to select.
     * @param mul The multiplier.
     * @return The maximum total sum of k selected elements.
     */
    public long maxSum(int[] nums, int k, int mul) {
        Arrays.sort(nums);
        long totalSum = 0;
        int currentMul = mul;
        int index = nums.length - 1;
        for (int i = 0; i < k; i++) {
            long effectiveMul = Math.max(currentMul, 1);
            totalSum += (long) nums[index] * effectiveMul;
            currentMul--;
            index--;
        }
        return totalSum;
    }
}